PowerEditor/src/ScintillaComponent/ScintillaEditView.cpp

```diff
@@ -3846,25 +3846,17 @@ void ScintillaEditView::sortLines(size_t fromLine, size_t toLine, ISorter* pSort
 	const auto endPos = execute(SCI_POSITIONFROMLINE, toLine) + execute(SCI_LINELENGTH, toLine);
 	const generic_string text = getGenericTextAsString(startPos, endPos);
 	std::vector<generic_string> splitText = stringSplit(text, getEOLString());
-	const size_t lineCount = execute(SCI_GETLINECOUNT);
-	const bool sortEntireDocument = toLine == lineCount - 1;
-	if (!sortEntireDocument)
+	bool lastLineEmpty = splitText.rbegin()->empty();
+	if (lastLineEmpty)
 	{
-		if (splitText.rbegin()->empty())
-		{
-			splitText.pop_back();
-		}
+		splitText.pop_back();
 	}
 	assert(toLine - fromLine + 1 == splitText.size());

```

Comment:
I am having a *really hard time* reconfiguring these debug assertions such that (a) they never fire and (b) everything behaves as intended.

I won't be able to look at this again for a few hours at least. Hopefully I will be able to avoid triggering debug assertions soon.

Reply:
I don't think you'll get spurious debug warnings any more. I tested it on all sorts, both ascending and descending, selecting the entire document, selecting part of the document, and not selecting anything.